Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -293,21 +306,46 @@ private Map<String, String> deriveResourceInfo(
                         resourceId = config.getDefaultResourceIdPrefix() + UUID.randomUUID();
                         resourceName = roleToResourceCount.getKey();
                     }
-
                     resouceNameToId.put(resourceName, resourceId);
 
                     return Resource.builder()
                         .id(resourceId)
                         .name(resourceName)
                         .totalAmount(roleToResourceCount.getValue())
                         .timetableId(config.getCustomTimetableId())
+                        .costPerHour(costingData.get(resourceName) == null ? 0 : costingData.get(resourceName))
                         .build();
                 }).collect(Collectors.toList()));
         }
 
         return resouceNameToId;
     }
 
+
+    private Map<String, Double> retrieveCostData(int logId) {
+        Map<String, Double> costRateMap = new HashMap<>();
+        try {
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            List<Usermetadata> userMetadata =
+                new ArrayList<>(userMetadataService.getUserMetadataByLog(logId, UserMetadataTypeEnum.COST_TABLE));
+
+            if (userMetadata.isEmpty()) {
+                return costRateMap;
+            }
+            CostingData[] costingDataList =
+                objectMapper.readValue(userMetadata.get(0).getContent(), CostingData[].class);
+            if (costingDataList != null && costingDataList.length > 0
+                && costingDataList[0].getCostRates() != null) {
+                costRateMap = costingDataList[0].getCostRates();
+            }
+        } catch (JsonProcessingException ex) {
+            log.error("Error in json data parsing", ex);
```

Comment:
Might be better to be a bit more specific with the error message. i.e. "Error in parsing cost json data"